### PR TITLE
Refactor storage tests - part 2

### PR DESCRIPTION
**Pull request by wojtek-t**
Ref https://github.com/kubernetes/kubernetes/issues/109831

This further refactors the package to build a set of generic storage tests.

```release-note
NONE
```

/sig api-machinery
/kind cleanup
/priority important-longterm

/assign @stevekuznetsov

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/storage/testing/utils.go [hunk 1]
```diff
@@ -232,10 +232,8 @@ func (p *PrefixTransformer) TransformToStorage(ctx context.Context, data []byte,
 	return data, p.err
 }
 
-func (p *PrefixTransformer) GetReads() uint64 {
-	return atomic.LoadUint64(&p.reads)
-}
-
-func (p *PrefixTransformer) ResetReads() {
+func (p *PrefixTransformer) GetReadsAndReset() uint64 {
+	result := atomic.LoadUint64(&p.reads)
```
Comment (stevekuznetsov) [inline staging/src/k8s.io/apiserver/pkg/storage/testing/utils.go:236]: Shouldn't this be a `SwapUnit64`?

Comment (wojtek-t) [reply staging/src/k8s.io/apiserver/pkg/storage/testing/utils.go:236]: done - good catch
[although probably doesn't matter in any of the existing tests]

[Also fixed in another place in the third commit]

**Review by wojtek-t — COMMENTED** [review]
@stevekuznetsov - thanks for the review, PTAL

**Review by stevekuznetsov — COMMENTED** [review]
Sweet, this looks good!

/lgtm

**Pull request merged**